Source/NSWindow.m

```diff
@@ -2249,12 +2249,13 @@ - (void) setFrame: (NSRect)frameRect display: (BOOL)flag
     {
       frameRect.size.height = _minimumSize.height;
     }
-      
+
   /* Windows need to be constrained when displayed or resized - but only
      titled windows are constrained */
   if (_styleMask & NSTitledWindowMask)
     {
-      frameRect = [self constrainFrameRect: frameRect toScreen: [self screen]];
+      frameRect = [self constrainFrameRect: frameRect
+                                  toScreen: [self _screenForFrame: frameRect]];
```

Comment:
I can agree that this could be slow at some degree (how often window changes its frame?). But why is it wrong?

Reply:
Wrong in the sense that it makes the code more complicated. I see _screenForFrame: as an internal method that should only be used in very few places.

Reply:
I understand. Although there is a problem with `[self screen]` usage here. At this point new values for `_frame` are not set yet. `-screen` method makes its calculations using `_frame` ivar values. So basically current code is wrong: it determines wrong screen. I consider using `_screen` instead of `[self screen]` at this call. Otherwise we need to set `_frame` to `newFrame` before `-constrain...` call and revert after. What do you think?

Reply:
You are right, sorry I did overlook this case.

Reply:
I have a better solution for this. I'll make separate PR (this involves PR into libs-back also).